### PR TITLE
Add Context support

### PR DESCRIPTION
**Pull request by Krenodeno**
Related to #31

### CODE REVIEW

**Comment by Krenodeno** [comment]
TODO: add support for context based gettext functions in the update module, and its tests

File: src/extract.js [hunk 1]
```diff
@@ -49,6 +51,18 @@ extract.main = function(jsFiles, output, options, callback) {
...
+    }
+    else if (typeof options.contextFunctions == "string") {
+        options.contextFunctions = options.contextFunctions.split(",");
+    }
+    if (options.pluralContextFunctions === undefined) {
+        options.pluralContextFunctions = ["npgettext", "lazyNpgettext"];
+    }
+    else if (typeof options.pluralContextFunctions == "string") {
+        options.pluralContextFunctions = options.pluralContextFunctions.split(",");
+    }
```
Comment (jbghoul) [inline src/extract.js:65]: Weird behaviour with a noop form extracted as context form

```js
const { lazyGettext: _, gettext_noop: N_, lazyPgettext } = require("stonejs");


webGLWarningBoxText: N_("Your browser does not support webGL. You won't be able to use 3D mode."),
```


image

**Comment by jbghoul** [comment]
My .po files are not updated with msgtxt

File: README.md [hunk 2]
```diff
@@ -29,11 +29,14 @@ To extract translatable strings from your Javascript and HTML sources, you shoul
     stonejs extract [options] <source js/html> <output pot>
 
 * `<source js/html>`: all Javascript and HTML files from which translatable string will be extracted (globbing allowed)
-* `<output pot>: the output translation template (`.pot`)
+* `<output pot>`: the output translation template (`.pot`)
```
Comment (jbghoul) [inline README.md:32]: my .po files are not updated with msgctx

Comment (Krenodeno) [reply README.md:32]: Do you any string that have a context ? string translated with `gettext` and `ngettext` does not have a `msgctxt` in the `.pot` after extraction

**Comment by Krenodeno** [comment]
- [ ] Fix multiple context for one string, extracting only one context

File: src/extract.js (hunk 1 above)
Comment (Krenodeno) [reply src/extract.js:65]: Okay, got it, we didn't cleared flags on ast marching. It should be good now.

**Comment by Krenodeno** [comment]
- [ ] Add context support to the `update` module

File: README.md (hunk 2 above)
Comment (Krenodeno) [reply README.md:32]: Resolved, it was because the `update` command did not support context-based strings.

**Pull request merged**